scripts/check_hook_registration.py

```
@@ -34,17 +37,34 @@ def registered_commands():
     return commands
 
 
+def gate_exit_code_raw(raw_text, working_dir):
+    """Run test-gate.py with arbitrary raw stdin (malformed-payload cases)."""
+    environment = dict(os.environ)
+    environment["GIT_CONFIG_GLOBAL"] = os.devnull
+    environment["GIT_CONFIG_SYSTEM"] = os.devnull
+    try:
+        result = subprocess.run([sys.executable, TEST_GATE], input=raw_text,
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)